src/bundle/Resources/public/ts/components/Badge.ts

```diff
@@ -0,0 +1,78 @@
+import Base from '../shared/Base';
+
+const THRESHOLD = {
+    medium: 100,
+    small: 10,
+};
+
+export default class BaseBadge extends Base {
+    private _value = '';
+    private _maxBadgeValue: number;
+
+    constructor(container: HTMLDivElement) {
+        super(container);
+
+        const _maxBadgeValue = this._container.dataset.ids_MaxBadgeValue;
+
+        if (!_maxBadgeValue || isNaN(parseInt(_maxBadgeValue, 10))) {
+            throw new Error('There is no proper max badge value defined for this badge!');
+        }
+        const parsedMax = parseInt(_maxBadgeValue, 10);
+
+        this._maxBadgeValue = parsedMax;
+
+        const initialValue = this._parseValue(this._container.textContent);
+
+        if (initialValue === null) {
+            throw new Error('No value found for this badge!');
+        }
+
+        this.setValue(initialValue);
+    }
+
+    private _parseValue(text: string | null): number | null {
```

Comment:
now I've started to wonder, `text` is quite generic term, maybe should be more like "badgeContent"?